### ISSUE TITLE
Support Typescript

### ISSUE THREAD

**Issue by IlyaSemenov** [lydell/eslint-plugin-simple-import-sort#3]
Currently the plugin crashes with `typescript-eslint-parser`:

```
TypeError: Expected `index` to be a number.
    at SourceCode.getLocFromIndex (/Users/semenov/tmp/ts-rule/node_modules/eslint/lib/util/source-code.js:436:19)
    at maybeReportSorting (/Users/semenov/tmp/ts-rule/node_modules/eslint-plugin-simple-import-sort/src/sort.js:62:27)
    at Program (/Users/semenov/tmp/ts-rule/node_modules/eslint-plugin-simple-import-sort/src/sort.js:20:11)
    at listeners.(anonymous function).forEach.listener (/Users/semenov/tmp/ts-rule/node_modules/eslint/lib/util/safe-emitter.js:45:58)
    at Array.forEach (<anonymous>)
    at Object.emit (/Users/semenov/tmp/ts-rule/node_modules/eslint/lib/util/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/Users/semenov/tmp/ts-rule/node_modules/eslint/lib/util/node-event-generator.js:251:26)
    at NodeEventGenerator.applySelectors (/Users/semenov/tmp/ts-rule/node_modules/eslint/lib/util/node-event-generator.js:280:22)
    at NodeEventGenerator.enterNode (/Users/semenov/tmp/ts-rule/node_modules/eslint/lib/util/node-event-generator.js:294:14)
    at CodePathAnalyzer.enterNode (/Users/semenov/tmp/ts-rule/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:632:23)
```

I created the reproduction repo: https://github.com/IlyaSemenov/eslint-plugin-simple-import-sort-typescript-repro

**Comment by lydell** [comment]
I'm not very familiar with TypeScript, but have gotten more interested in it recently.

As a first step – TypeScript has some custom `import` syntax, other than the standard ES2015 syntax, or am I mistaken? If so, do you know of some place that lists them all?

**Comment by IlyaSemenov** [comment]
Not at all, the syntax is exactly the same: https://www.typescriptlang.org/docs/handbook/modules.html#import

**Comment by IlyaSemenov** [comment]
Right, technically you are correct although I've never seen this syntax being used in wild.

As I understood, this is a counterpart of `const foo = require('foo')` which also loads typing information, so it's not really an import. Since your plugin officially does not support `require` I assume it's safe to ignore this type of import syntax as well.

**Comment by lydell** [comment]
Cool, so maybe all we have to do then is to add test cases with the Typescript parser and track down why it crashes.

**Comment by lydell** [comment]
Released in v2.1.0. Thanks for the help!

**Comment by IlyaSemenov** [comment]
This is super awesome, I'm using it already. Thanks so much for publishing.
